packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js

```diff
@@ -171,76 +171,67 @@ export default class ReconnectionManager {
 
     LoggerProxy.logger.info('ReconnectionManager:index#validate --> Reconnection is not enabled.');
 
-    return new ReconnectionError('Reconnection is not enabled.');
+    throw new ReconnectionError('Reconnection is not enabled.');
   }
 
   /**
    * Initiates a media reconnect for the active meeting
    * @param {Object} reconnectOptions
-   * @param {boolean} [reconnectOptions.networkChange=false] indicates if a network change event happened
    * @param {boolean} [reconnectOptions.networkDisconnect=false] indicates if a network disconnect event happened
    * @returns {Promise}
    * @public
    * @memberof ReconnectionManager
    */
-  reconnect({networkChange = false, networkDisconnect = false} = {}) {
-    const validatedReconnect = this.validate();
-
-    if (validatedReconnect === true) {
-      Metrics.postEvent({
-        event: eventType.MEDIA_RECONNECTING,
-        meeting: this.meeting
-      });
-
-      return this.executeReconnection({networkChange, networkDisconnect})
-        .then(() => {
-          Metrics.postEvent({
-            event: eventType.MEDIA_RECOVERED,
-            meeting: this.meeting
-          });
-        })
-        .catch((reconnectError) => {
-          if (reconnectError instanceof NeedsRetryError) {
-            // Reset our reconnect status since we are looping back to the beginning
-            this.status = RECONNECTION.STATE.DEFAULT_STATUS;
-
-            return this.reconnect({networkDisconnect: true});
-          }
+  async reconnect({networkDisconnect = false} = {}) {
+    // First, validate that we can reconnect, if not, it will throw an error
+    this.validate();
+
+    Metrics.postEvent({
+      event: eventType.MEDIA_RECONNECTING,
+      meeting: this.meeting
+    });
+
+    return this.executeReconnection({networkDisconnect})
+      .then(() => {
+        Metrics.postEvent({
+          event: eventType.MEDIA_RECOVERED,
+          meeting: this.meeting
+        });
+      })
+      .catch((reconnectError) => {
+        if (reconnectError instanceof NeedsRetryError) {
+          // Reset our reconnect status since we are looping back to the beginning
+          this.status = RECONNECTION.STATE.DEFAULT_STATUS;
 
-          Metrics.postEvent({
-            event: eventType.MEDIA_RECONNECTION_FAILED,
-            meeting: this.meeting
-          });
+          return this.reconnect({networkDisconnect: true});
+        }
 
-          throw reconnectError;
+        Metrics.postEvent({
+          event: eventType.MEDIA_RECONNECTION_FAILED,
+          meeting: this.meeting
         });
-    }
 
-    return Promise.reject(validatedReconnect);
+        throw reconnectError;
+      });
   }
 
   /**
    * @param {Object} reconnectOptions
-   * @param {boolean} [reconnectOptions.networkChange=false] indicates if a network change event happened
    * @param {boolean} [reconnectOptions.networkDisconnect=false] indicates if a network disconnect event happened
    * @returns {Promise}
    * @private
```

Comment:
Does this still return a promise? Should we add JSDoc for the `throws` here?

Reply:
async function always returns a promise. Will add the throws here.